### ISSUE TITLE
Question Set 1

### ISSUE THREAD

**Issue by andrewniehaus** [andrewniehaus/adleoAN#1]
adleo questions #1

Q1. 
1. A 100x100 greyscale image with 1000 nodes and 10 output categories would have 10,000 trainable parameters in the first layer connecting to the hidden features (1000 nodes X 10 output categories) and 10,000,000 parameters as the hidden layer considers each other node in mapping to output classes. 

2. The same image but with 7 bands would have 70,000 trainable parameters and____ in the first layer and 70,000,000 trainable parameters in the second layer mapping to output classes. 

Q2. Non linear activation functions allow for models to work with complex dynamics. In loss functions, minima can be found in the non linear functions that are past local minima through gradient descent.

Q3. ReLU is preferred as an activation function over sigmoid functions or tanh because it limits exploding gradients and is more efficient with inputs. Given that it zeros out any values below 0 and has a direct variation for values >0, progressive increases or decreases in values in the ReLu function do not lead to exploding gradients. 

Q4. Shuffling is uncommon in validation and test datasets because its primary purpose is to prevent the model from getting stuck on local minima as it runs through gradient descent. In validation and testing, this is not a concern thus eliminating the need for shuffling.

**Comment by Rahebe22** [comment]
Good job :) 
Here are just some modifications:

Q1. part 1: first layer: (100*100  *  1000) weights + 1000 biases = 10001000

part 2: first layer: (7*100*100  *1000) weights + 1000 biases = 70001000

Q2. "If we did not add nonlinear activation functions, the model would essentially only be a weighted average of the inputs."

Q3.  In Sigmoid, the derivative reduces to zero as inputs become very large (**both positive and negative**) which is called "vanishing gradient problem".
